src/corehost/cli/fxr/hostfxr.cpp

```diff
@@ -99,3 +99,67 @@ SHARED_API int hostfxr_main(const int argc, const pal::char_t* argv[])
     return muxer.execute(argc, argv);
 }
 
+//
+// Determines the directory location of the SDK accounting for global.json and multi-level lookup policy.
+//
+// Invoked via MSBuild SDK resolver to locate SDK props and targets from an msbuild other than the one
+// bundled by the CLI.
+//
+// Parameters:
+//    global_json_start_dir - The directory where the search for global.json 
+//    buffer_size           - The size of the buffer argument in pal::char_t units (see encoding note below)
+//    buffer               -  The buffer where the located SDK path will be written.
+//
+// Return value:
+//   <0 - Invalid argument
+//   0  - SDK could not be found.
+//   >0 - The number of characters (including null terminator) required to store the located SDK.
+//
+//   If resolution succeeds and the positive return value is less than or equal to buffer_size (i.e. the
+//   the buffer is large enough), then the resolved SDK path is copied to the buffer and null terminated.
+//   Otherwise, no data is written to the buffer.
+//
+// String encoding:
+//   Windows     - UTF-16 (pal::char_t is 2 byte wchar_t)
```

Comment:
I thought a lot about this and I think it's wasteful to convert unnecessarily. There's currently no way to P/Invoke with automatic UTF-8 marshaling on Windows so that makes UTF-8 a cumbersome standard choice when our only use case is from Windows. This is the simplest implementation and the simplest portable P/invoke us actually to query RuntimeInformation and use two different DllImports with varying CharSet. 

If we have a need in the future for a fixed encoding across platforms, we can add _utf8 or _utf16 suffixed exports, but I'm not seeing that need in practice.